src/pages/intent/avatar.tsx

```diff
@@ -0,0 +1,179 @@
+import { Button, Center, Container, Group, Loader, Paper, Stack, Text, Title } from '@mantine/core';
+import { useRouter } from 'next/router';
+import React, { useEffect, useState } from 'react';
+import { z } from 'zod';
+import { UserAvatar } from '~/components/UserAvatar/UserAvatar';
+import { useCFImageUpload } from '~/hooks/useCFImageUpload';
+import { useCurrentUser } from '~/hooks/useCurrentUser';
+import { constants } from '~/server/common/constants';
+import { createServerSideProps } from '~/server/utils/server-side-helpers';
+import { loadImage } from '~/utils/blurhash';
+import { formatDate } from '~/utils/date-helpers';
+import { getLoginLink } from '~/utils/login-helpers';
+import { showSuccessNotification } from '~/utils/notifications';
+import { formatBytes } from '~/utils/number-helpers';
+import { trpc } from '~/utils/trpc';
+
+const avatarQuerySchema = z.object({
+  mediaUrl: z.string().url(),
+});
+
+export const getServerSideProps = createServerSideProps({
+  useSSG: true,
+  useSession: true,
+  resolver: async ({ session, ctx }) => {
+    if (!session) {
+      return {
+        redirect: {
+          destination: getLoginLink({ returnUrl: ctx.resolvedUrl, reason: 'perform-action' }),
+          permanent: false,
+        },
+      };
+    }
+    if (session.user?.muted) return { notFound: true };
+  },
+});
+
+export default function IntentAvatar() {
+  const router = useRouter();
+  const [loading, setLoading] = useState(true);
+  const [waiting, setWaiting] = useState(false);
+  const [errorMsg, setErrorMsg] = useState<React.ReactNode | undefined>(undefined);
+  const [imgToUpload, setImgToUpload] = useState<File>();
+  const currentUser = useCurrentUser();
+  const { uploadToCF } = useCFImageUpload();
+
+  const queryParse = avatarQuerySchema.safeParse(router.query);
+  const url = queryParse.success ? queryParse.data.mediaUrl : undefined;
+
+  const { mutate } = trpc.user.update.useMutation({
+    async onSuccess() {
+      currentUser?.refresh();
+      router.push('/user/account').then(() => {
+        showSuccessNotification({ message: 'Your avatar has been updated.' });
+        setWaiting(false);
+      });
```

Comment:
An alternative for cases where a user is setting their avatar from an external url (avatar-gen)
 ```suggestion
     if(document.referrer.length) {
        showSuccessNotification({ message: 'Your avatar has been updated.' });
        setTimeout(() => {
          router.push(document.referrer)
        }, 3000);
      } else {
        router.push('/user/account').then(() => {
          showSuccessNotification({ message: 'Your avatar has been updated.' });
          setWaiting(false);
        });
      }
```